File: src/main/java/ar/edu/unq/epers/unidad5/service/ProductoService.java
package ar.edu.unq.epers.unidad5.service;

import ar.edu.unq.epers.unidad5.model.Producto;
import ar.edu.unq.epers.unidad5.model.PrecioPromedio;
import java.util.List;

public interface ProductoService {
    Producto save(Producto producto);
    void saveAll(List<Producto> productos);
    List<Producto> findByBrand(String marca);
    Producto getByCode(String code);
    List<Producto> findByPrice(int precio);
    List<Producto> findByPriceRange(int min, int max);
    List<PrecioPromedio> getPrecioPromedio(List<String> codigos);
    void deleteAll();
}

File: src/main/kotlin/ar/edu/unq/epers/unidad5/service/ProductoService.kt
package ar.edu.unq.epers.unidad5.service

import ar.edu.unq.epers.unidad5.model.PrecioPromedio
import ar.edu.unq.epers.unidad5.model.Producto

interface ProductoService {
    fun save(producto: Producto): Producto
    fun saveAll(productos: List<Producto>)
    fun findByBrand(marca: String): List<Producto>
    fun getByCode(code: String): Producto
    fun findByPrice(precio: Int): List<Producto>
    fun findByPriceRange(min: Int, max: Int): List<Producto>
    fun getPrecioPromedio(codigos: List<String>): List<PrecioPromedio>
    fun deleteAll()
}